packages/bus/src/Brokers/RMQEventBus.ts

```
@@ -0,0 +1,395 @@
+
+      return response as R;
+    } finally {
+      this.subject.removeAllListeners(correlationId);
+    }
+  }
+
+  public async destroy(): Promise<void> {
+    if (!this.client) {
+      return;
+    }
+
+    if (this.channel) {
+      await this.channel.waitForConfirms();
+      await Promise.all(
+        this.consumerTags.map(consumerTag => this.channel?.cancel(consumerTag))
+      );
+      await this.channel.close();
+    }
+
+    await this.client.close();
+    this.clear();
+  }
+
+  private getWaiter<R>(ttl: number, correlationId: string): Promise<[R]> {
+    return Promise.race([
+      once(this.subject, correlationId) as Promise<[R]>,
+      new Promise<never>((_, reject) =>
+        setTimeout(reject, ttl, new NoResponse(ttl))
+      )
+    ]);
+  }
+
+  private reflectEventsNames<T, R>(
+    handler: EventHandlerConstructor<T, R>
+  ): string[] {
+    const types: (new (...args: unknown[]) => Event<T> | string)[] =
+      Reflect.getMetadata(Event, handler) ?? [];
+
+    return types.map((event: new (...args: unknown[]) => Event<T> | string) =>
+      typeof event === 'string' ? event : event.name
+    );
+  }
+
+  private async createConsumerChannel(): Promise<void> {
+    if (this.channel) return;
+
+    if (!this.client) {
+      throw new Error('The client was not created.');
+    }
+
+    this.channel = await this.client.createConfirmChannel();
+
+    await this.bindExchangesToQueue();
+    await this.startBasicConsume();
+    await this.startReplyQueueConsume();
+  }
+
+  private async startReplyQueueConsume(): Promise<void> {
+    if (!this.channel) {
+      return;
+    }
+
+    const { consumerTag } = await this.channel.consume(
+      this.REPLY_QUEUE_NAME,
+      (msg: ConsumeMessage | null) => this.processReply(msg),
+      {
+        noAck: true
+      }
+    );
+
+    this.consumerTags.push(consumerTag);
+  }
+
+  private async startBasicConsume(): Promise<void> {
+    if (!this.channel) {
+      return;
+    }
```

You should raise an exception